include/ensmallen_bits/cmaes/cmaes_impl.hpp

```diff
@@ -313,6 +324,90 @@ typename MatType::elem_type CMAES<SelectionPolicyType>::Optimize(
   return overallObjective;
 }
 
+// Transforms the candidate into the given bounds.
+template<typename SelectionPolicyType>
+template<typename BaseMatType>
+void CMAES<SelectionPolicyType>::BoundaryTransform(BaseMatType& matrix)
```

Comment:
So I dived into the code and I think it makes sense to introduce another policy e.g. `TransformationPolicy` so we can enable/disable the boundary constraint. Let me know what you think, I can push the changes to you branch, if you like?

Reply:
Hi, @zoq thanks for the review. I will implement the `TransformationPolicy` this week.

Reply:
So basically it's similar to how the update policies are implemented https://github.com/mlpack/ensmallen/tree/master/include/ensmallen_bits/sgd/update_policies.

We basically add a new template type called `TransformationPolicy` or something like that and define an interface. I think we just need two functions at this point `Transform` and `Inverse`. So the class policy class for the `BoundaryBoxConstraint` could look like:

```c++
class BoundaryBoxConstraint
{
 public:
  void Transform(...)
  {
  }

  void Inverse(...)
  {
  }
}
```

we also a policy class that does nothing:

```c++
class EmptyTransformation
{
 public:
  void Transform(...)
  {
  }

  void Inverse(...)
  {
  }
}
```


in the `CMAES` class we can than use an instantiation of the `TransformationPolicy` and call the `Transform` and `Inverse` methods. Hopefully what I wrote above makes sense, happy to clarify anything.